src/genie/libs/parser/asa/show_route.py

```diff
@@ -51,6 +51,33 @@ class ShowRouteSchema(MetaParser):
                                 },
                             },
                         },
+                        Optional('tunneled_routes'): {
```

Comment:
Since Takashi was able to find a nice solution by separating routes into `'routes'` and `'tunneled_routes'` and since the two are identical except for name, we can now leverage a handy schema tool called `Or()`.

```
from genie.metaparser.util.schemaengine import Any, Optional, Or
...
class ShowRouteSchema(MetaParser):
    """Schema for
        * show route
    """
    schema = {
        'vrf': {
            'default': {
                'address_family': {
                    'ipv4': {
                        Optional(Or('routes', 'tunneled_routes')): {
                            Any(): {
                                'candidate_default': bool,
                                Optional('subnet'): str,
                                'route': str,
                                Optional('active'): bool,
                                Optional('date'): str,
                                Optional('route_preference'): int,
                                Optional('metric'): int,
                                Optional('source_protocol'): str,
                                Optional('source_protocol_codes'): str,
                                Optional('next_hop'): {
                                    Optional('outgoing_interface_name'): {
                                        Any(): {  # context_name for interface if there is no next_hop
                                            Optional('outgoing_interface_name'): str
                                        },
                                    },
                                    Optional('next_hop_list'): {
                                        Any(): {  # index
                                            Optional('index'): int,
                                            Optional('next_hop'): str,
                                            Optional('outgoing_interface_name'): str
                                        },
                                    },
                                },
                            },
                        },
                    },
                },
            },
        },
    }
```
This should work and will cut down on extra keys

Note: This isn't a required change since what you have currently works fine. Only implement this idea if you'd like to

Reply:
I like the smaller change, pushing update.